src/derivative_utils.jl

```diff
@@ -147,6 +157,10 @@ function Base.:\(W::WOperator, x::Union{AbstractVecOrMat,Number})
   end
 end
 function LinearAlgebra.mul!(Y::AbstractVecOrMat, W::WOperator, B::AbstractVecOrMat)
+  if W._func_cache == nothing
+    # Allocate cache only if needed
+    W._func_cache = Vector{eltype(W)}(undef, size(Y, 1))
```

Comment:
I think algorithm caches should be able to supply this during construction when possible. That's an optimization we can add later.